src/test/java/com/jetbulb/interviews/NotationConverterTest.java

```diff
@@ -0,0 +1,50 @@
+package com.jetbulb.interviews;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.NullAndEmptySource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowableOfType;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+public class NotationConverterTest {
+    private final NotationConverter converter = new NotationConverter();
+
+    @ParameterizedTest
+    @NullAndEmptySource
+    @ValueSource(strings = {"A", "1", "X1", "_", "Vi1"})
+    void shouldThrowInvalidValueExceptionIfNonRomanDigitFound(String in) {
+        assertThat(
+                catchThrowableOfType(() -> converter.toArabic(in), InvalidValueException.class)
+        )
+                .overridingErrorMessage("InvalidValueException should be thrown if invalid symbols detected")
+                .isNotNull()
+                .hasMessage("String must contain only valid roman numerals [I, V, X, L, C, D, M]");
+    }
+
+    @ParameterizedTest
+    @MethodSource("romanAndArabicNotationProvider")
+    void shouldSuccessfullyConvertRomanToArabicNotation(String roman, int arabic) {
+        assertThat(converter.toArabic(roman))
+                .overridingErrorMessage(
+                        "Should be arabic value %s after conversion from roman notation %s",
+                        arabic,
+                        roman
+                )
+                .isEqualTo(arabic);
+    }
+
+    private static Stream<Arguments> romanAndArabicNotationProvider() {
```

Comment:
What do you think about introducing TestData Object with all the scenario data in it instead having many arguments?
```
TestData(String roman, int arabic)

then use like this
private static Stream<TestData> romanAndArabicNotationProvider()
```